src/jaxsim/api/references.py

```diff
@@ -441,3 +442,111 @@ def convert_using_link_frame(
+
+        # If we have the model, we can extract the frame names if not provided.
+        frame_names = frame_names if frame_names is not None else model.frame_names()
+
+        # Make sure that the frame names are a tuple if they are provided by the user.
+        frame_names = (frame_names,) if isinstance(frame_names, str) else frame_names
+
+        if len(frame_names) != f_F.shape[0]:
+            msg = "The number of frame names ({}) must match the number of forces ({})"
+            raise ValueError(msg.format(len(frame_names), f_F.shape[0]))
+
+        # Extract the frame indices.
+        frame_idxs = js.frame.names_to_idxs(frame_names=frame_names, model=model)
+        parent_link_idxs = js.frame.idx_of_parent_link(
+            model=model, frame_idxs=frame_idxs
+        )
+
+        if not_tracing(forces) and not data.valid(model=model):
+            raise ValueError("The provided data is not valid for the model")
+
+        W_H_Fi = jax.vmap(
+            lambda frame_idx: js.frame.transform(
+                model=model, data=data, frame_index=frame_idx
+            )
+        )(frame_idxs)
+
+        # Helper function to convert a single 6D force to the inertial representation
+        # considering as body the frame (i.e. L_f_F and LW_f_F).
+        def to_inertial(
+            f_F: jtp.MatrixLike, W_H_F: jtp.MatrixLike, frame_idx: jtp.ArrayLike
+        ) -> jtp.Matrix:
+            return JaxSimModelReferences.other_representation_to_inertial(
+                array=f_F,
+                other_representation=self.velocity_representation,
+                transform=W_H_F,
+                is_force=True,
+            )
+
+        match self.velocity_representation:
+            case VelRepr.Inertial:
+                W_f_F = f_F
+
+            case VelRepr.Body | VelRepr.Mixed:
+                W_f_F = jax.vmap(to_inertial)(f_F, W_H_Fi, frame_idxs)
+
+            case _:
+                raise ValueError("Invalid velocity representation.")
+
+        def convert_to_link_force(
+            W_f_F: jtp.MatrixLike, W_H_F: jtp.MatrixLike, parent_link_idx: jtp.ArrayLike
+        ) -> jtp.Matrix:
+            W_H_L = js.link.transform(
+                model=model, data=data, link_index=parent_link_idx
+            )
+
+            F_H_L = js.transform.inverse(W_H_F) @ W_H_L
+            F_X_L = Adjoint.from_transform(transform=F_H_L, inverse=True)
+
+            return F_X_L @ W_f_F
+
+        W_f_L_i = jax.vmap(convert_to_link_force)(W_f_F, W_H_Fi, parent_link_idxs)
+
+        # Sum the forces on the parent links.
+        mask = parent_link_idxs[:, jnp.newaxis] == jnp.arange(model.number_of_links())
+        W_f_L = mask.T @ W_f_L_i
+
+        return self.apply_link_forces(
+            model=model,
+            data=data,
+            link_names=js.link.idxs_to_names(
+                model=model, link_indices=parent_link_idxs
+            ),
+            forces=W_f_L,
+            additive=additive,
+        )
```

Comment:
Since you're passing inertial-fixed forces, you need to use `self` within a context manager that switches to `VelRepr.Inertial`.

Reply:
You're right! Done in https://github.com/ami-iit/jaxsim/pull/209/commits/cf74cfffcd101e1d08f802c03e14ec6f078e9129